src/c/core/serialization/xrce_types.c

```diff
@@ -271,18 +271,22 @@ bool uxr_serialize_TransportLocatorSeq(ucdrBuffer* buffer, const TransportLocato
 bool uxr_deserialize_TransportLocatorSeq(ucdrBuffer* buffer, TransportLocatorSeq* output)
 {
     bool ret = ucdr_deserialize_uint32_t(buffer, &output->size);
-    if(output->size > UXR_TRANSPORT_LOCATOR_SEQUENCE_MAX)
-    {
-        buffer->error = true;
-        ret = false;
-    }
-    else
+
+    // If locator seq does not fit, only uses the first ones.
+    // This is the case in Agent discovery, take care if this funtion
+    // is used elsewhere.
+
+    output->size = (output->size > UXR_TRANSPORT_LOCATOR_SEQUENCE_MAX)   ?
+                        UXR_TRANSPORT_LOCATOR_SEQUENCE_MAX               :
+                        output->size;
+
+    for(uint32_t i = 0; i < output->size && ret; i++)
     {
-        for(uint32_t i = 0; i < output->size && ret; i++)
-        {
-            ret = uxr_deserialize_TransportLocator(buffer, &output->data[i]);
-        }
+        ret = uxr_deserialize_TransportLocator(buffer, &output->data[i]);
     }
+
+    buffer->error =  (output->size > UXR_TRANSPORT_LOCATOR_SEQUENCE_MAX) ? true : false;
+
     return ret;
```

Comment:
```suggestion
    return ret && !buffer->error;
```

Reply:
If we return `false` here the complete deserialization fails and the discovery system does not recognize the incoming message [here](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/66df0a6df20063d246dd638ac3d33eb2e652fab2/src/c/profile/discovery/discovery.c#L145) and does not call the user callback.

This is the point of the tweak... maybe we should prepare a discovery dedicated deserialization function that takes this into account instead of changing the generic `uxr_deserialize_TransportLocatorSeq`.

Reply:
ok, I understand it now.  Yes, I think that would be a better approach; changing the default behaviour of that function can lead to dangerous situations, specially if you are using the method's boolean result as a conditional instead of the `buffer->error` variable.

Reply:
Reverted, please check new changes.